### ISSUE TITLE
Add Links to WCAG SC

### ISSUE THREAD

**Issue by mgifford** [w3c/wai-arrm#49]
I was just showing this page to one of our designers:
https://www.w3.org/WAI/planning/arrm/visual-designer/

It occurred to me that it would be useful to have links as I know he doesn't know what 1.1.1 is.

In talking to @shawna-slh and @remibetin it made sense to add a link to point folks to the SC in the Quick: https://www.w3.org/WAI/WCAG22/quickref/

Although pointing to:
https://www.w3.org/WAI/WCAG21/Understanding/non-text-content

is way less scary. 

I don't know how to go from 1.1.1 to https://www.w3.org/WAI/WCAG21/Understanding/non-text-content though.

Ideally there would be a redirect:

https://www.w3.org/WAI/WCAG21/Understanding/1.1.1

Same issue with:
https://www.w3.org/WAI/WCAG22/quickref/#non-text-content

Would be nice to point to:

https://www.w3.org/WAI/WCAG22/quickref/#1.1.1

**Comment by mgifford** [comment]
I think we can do something like this:

```
{% assign sc_number = row["WCAG SC"] %}
{% assign wcag_entry = site.data.wcag | where: "key", sc_number | first %}
<td>
  {% if wcag_entry %}
    <a href="https://www.w3.org/WAI/WCAG22/Understanding/{{ wcag_entry.slug | uri_escape }}">{{ sc_number }}</a>
  {% else %}
    {{ sc_number }}
  {% endif %}
</td>
```

Leveraging _data/wcag.yml

To quickly add links to the understanding documents.

**Comment by shawna-slh** [comment]
For more input: Adding links would add some complexity, perhaps especially for screen reader users. It would be good to get input on usability and perspectives from several people.

**Comment by dboudreau** [comment]
We had these links to the understanding document back when this info was in the wiki, so it makes sense to bring it back in these pages. This issue was discussed in the co-chair meeting and we are all in favour of doing this.

**Comment by mgifford** [comment]
I'll look at adding the aria-label to the title of the SC that is specified, as this would be an improvement. 

So the aria-label includes **key** & **title** so that it is "1.1.1: Non-text Content"

@shawna-slh good point about https://github.com/w3c/wai-arrm/issues/49#issuecomment-2783565846 @kfranqueiro any ETA on the new WCAG JSON? We will need the update for when we do WCAG 2.2. 

@shawna-slh a future enhancement for the W3C is suggested to be the [aria tooltip](https://www.w3.org/WAI/ARIA/apg/patterns/tooltip/). Would this be a possibility?

**Comment by kfranqueiro** [comment]
RE _data/wcag.yml:

This file originates in the [wai-website-data repository](https://github.com/w3c/wai-website-data). As Shawn points out, it has not been actively maintained. Upon scrutiny, there are 2 problems that would make it unfit to reuse here:

- Its `slug` properties reference old IDs from WCAG 2.0, which do not match the Understanding URLs (but do match anchors in Quickref)
- It is missing several success criteria altogether

This data file is actively used for one specific page within the WAI website, which would be impacted if we were to update that file in-place.

RE wcag.json updates:

As of a couple of weeks ago, we have an [updated build process](https://github.com/w3c/wcag/pull/4301) for producing wcag.json files for WCAG 2.1 and 2.2, which have been published to the WAI website from the wcag repo. However, consuming this directly over the network in its full format would require far more effort.

The WAI team plans to discuss the best format for importing the updated data into the wai-website-data repo, to make it available similarly to the data you were working with above.

**Comment by iadawn** [comment]
@mgifford Would you be able to use the newly available [wcag.json](https://www.w3.org/WAI/WCAG22/wcag.json)?

This has everything in it and will be maintained. You should be able to grab the SC id and use that to construct a link to the understanding document?

**Comment by mgifford** [comment]
Any updates on this @kfranqueiro & @iadawn ? 

It would be good if there were more consistency among the id's.  Not that this is new for you, but maybe we need a id-legacy column on the JSON to help address the fact that this wasn't done before.

In looking at the build process, it looks like the .json file was deleted. At least it says "This file was deleted." https://github.com/w3c/wcag/blob/9fbbdb3a09a1b6fa71754cc6fcd15c6d38ac277f/guidelines/wcag.json

**Comment by kfranqueiro** [comment]
> In looking at the build process, it looks like the .json file was deleted. At least it says "This file was deleted." https://github.com/w3c/wcag/blob/9fbbdb3a09a1b6fa71754cc6fcd15c6d38ac277f/guidelines/wcag.json

Yes, the JSON file was removed from the source repository in the pull request I linked earlier, w3c/wcag#4301, because it is now part of the build output (not an input), and its presence in the source structure had caused confusion on multiple occasions.

> It would be good if there were more consistency among the id's. Not that this is new for you, but maybe we need a id-legacy column on the JSON to help address the fact that this wasn't done before.

This issue is unique to the several-years-old `wcag.yml` file in wai-website-data. Newer versions of `wcag.json` consistently point to the modern IDs under the `id` property (old IDs are under `alt_id` instead).

After discussing among the WAI Team, it became clear that `wcag.yml` is used on a few specific pages, and it is best to leave it intact; it has since been [marked as legacy](https://github.com/w3c/wai-website-data/commit/87c02ba68f650129cdfc3e9d4cb75e7197f76119). We plan to make updated data available to templates via a separate namespace.

> Any updates on this @kfranqueiro & @iadawn ?

After a couple of WAI Team discussions, we're mostly aligned on a way forward, give or take a couple of minor details; once I'm sure of those and have an opportunity to verify, I'll provide an update regarding the code block you posted earlier.

**Comment by mgifford** [comment]
That all seems to work fine https://deploy-preview-76--wai-arrm.netlify.app/planning/arrm/tasks/

PR - https://github.com/w3c/wai-arrm/pull/76
